src/AuthorizedSocketConnection.js

```diff
@@ -190,8 +190,12 @@ export default class AuthorizedSocketConnection<TContext, TCredentials> {
       try {
         resultOrStream = await sourcePromise;
       } catch (err) {
-        this.subscriptions.delete(id);
-        throw err;
+        if (err instanceof GraphQLError) {
+          resultOrStream = { errors: [err] };
```

Comment:
`createSourceEventStream` will occasionally just throw `GraphQLError`

Reply:
https://github.com/graphql/graphql-js/blob/4fd9260efa247d60f74f3ebc259c8a3c2b03af61/src/subscription/subscribe.js#L272-L274

these all get caught and wrapped from the subscribe function, but not from here